src/nbl/video/COpenGLFence.h

```diff
@@ -0,0 +1,78 @@
+#ifndef __NBL_C_OPENGL_FENCE_H_INCLUDED__
+#define __NBL_C_OPENGL_FENCE_H_INCLUDED__
+
+#include "nbl/video/IGPUFence.h"
+#include "nbl/video/IOpenGLSyncPrimitiveBase.h"
+#include "nbl/video/IOpenGL_FunctionTable.h"
+
+namespace nbl {
+namespace video
+{
+
+class COpenGLFence final : public IGPUFence, public IOpenGLSyncPrimitiveBase
+{
+protected:
+    ~COpenGLFence() = default;
+
+public:
+    // signaled ctor
+    COpenGLFence(IOpenGL_LogicalDevice* dev, ILogicalDevice* _dev, IOpenGL_FunctionTable* gl) : IGPUFence(_dev, ECF_SIGNALED_BIT), IOpenGLSyncPrimitiveBase(dev)
+    {
+        signal(gl);
+    }
```

Comment:
take a `smart_refctd_ptr<COpenGLSync>&&` instead, and use the `signal` variant that does not create its own sync object.